src/cmd/ksh93/tests/chmod.sh

```diff
@@ -0,0 +1,203 @@
+# Tests for `chmod` builtin.
+#
+# At the moment this is primarily to avoid a regression of issue #949.
+
+# chmod - change the access permissions of files
+
+#  chmod changes the permission of each file according to mode, which can be
+#  either a symbolic representation of changes to make, or an octal number
+#  representing the bit pattern for the new permissions.
+#
+
+# OPTIONS
+
+umask 077  # start by disabling setting all group/other permission bits by default
+mkdir foo
+ln -s foo symlink_to_foo
+touch foo/bar
+ln -s bar symlink_to_bar
+
+# Create some files to use in this test.
+umask 077  # start by disabling setting all group/other permission bits by default
+mkdir d
+touch a b d/e d/f
+
+# Create a function to report the pathname permissions in a fashion that works on BSD, Linux, and
+# other platforms. This could be simplified if ksh supported a `stat` builtin.
+if [[ $(stat -f '%Sp' a 2>/dev/null) == '-rw-------' ]]
+then
+    # This appears to be a BSD `stat` command that supports the `-f` flag.
+    function stat_perms {
+        stat -f '%Sp' "$1"
+    }
+elif [[ $(stat --format '%A' a 2>/dev/null) == '-rw-------' ]]
+then
+    # This appears to be a GNU `stat` command that supports the `-A` flag.
+    function stat_perms {
+        stat -c '%A' "$1"
+    }
+else
+    # We have no idea how to report pathname permissions in a human readable form.
+    log_error "no usable 'stat' command found"
```

Comment:
The way I've handled this type of situation in other unit tests is to use this pattern since there isn't any point running any of the test and this doesn't really represent a failure of the `chmod` builtin:

```
log_warning "no usable 'stat' command found"
exit 0
```